test/e2e/knative_serving_test.go

```diff
@@ -57,6 +65,75 @@ func TestKnativeServing(t *testing.T) {
 	})
 }
 
+func testKnativeVersusKubeServicesInOneNamespace(t *testing.T, caCtx *test.Context) {
+	// Deploy plain Kube service
+	svc, err := test.CreateKubeService(caCtx, kubeHelloworldService, testNamespace2, image)
+	if err != nil {
+		t.Fatal("Kubernetes service not created", err)
+	}
+	route, err := test.WithRouteForServiceReady(caCtx, svc.Name, testNamespace2)
+	if err != nil {
+		t.Fatal("Failed to create route for service", svc.Name, err)
+	}
+	kubeServiceURL := "http://" + route.Status.Ingress[0].Host
+
+	// Check Kube service responds
+	waitForRouteServingText(t, caCtx, kubeServiceURL, helloworldText)
+
+	// Deploy Knative service in the same namespace
+	ksvc, err := test.WithServiceReady(caCtx, helloworldService, testNamespace2, image)
+	if err != nil {
+		t.Fatal("Knative Service not ready", err)
+	}
+
+	// Check that both services respond
+	waitForRouteServingText(t, caCtx, ksvc.Status.URL.Host, helloworldText)
+	waitForRouteServingText(t, caCtx, kubeServiceURL, helloworldText)
+
+	// Delete Knative service
+	test.DeleteService(caCtx, ksvc.Name, testNamespace2)
+
+	// Check that Kube service still responds
+	waitForRouteServingText(t, caCtx, kubeServiceURL, helloworldText)
+
+	// Remove the Kube service
+	test.DeleteRoute(caCtx, svc.Name, testNamespace2)
+	test.DeleteKubeService(caCtx, svc.Name, testNamespace2)
```

Comment:
This helper should match the `CreateKubeService` helper and actually remove the deployment as well by the logic of the other helper.

Reply:
I've split the CreateKubeService even more, now the Deployment,Service,Route are deployed and undeployed separately.